### ISSUE TITLE
CloudBus: `if_nameindex` is not a member of `ACE_OS`

### ISSUE THREAD

**Issue by naegelejd** [gadgetron/gadgetron#115]
We're compiling the Gadgetron on OpenSUSE 13.2 and the compiler is complaining that both `if_nameindex` and `if_freenameindex` are not in the `ACE_OS` namespace. The compiler recommends to use the freestanding versions of the function provided by the system. Here's the output:

```
/home/rtadmin/RTafni/src/gadgetron/toolboxes/cloudbus/CloudBus.cpp: In member function ‘virtual int Gadgetron::CloudBusReceiverTask::open(void*)’:
/home/rtadmin/RTafni/src/gadgetron/toolboxes/cloudbus/CloudBus.cpp:43:12: error: ‘if_nameindex’ is not a member of ‘ACE_OS’
     intf = ACE_OS::if_nameindex ();
            ^
/home/rtadmin/RTafni/src/gadgetron/toolboxes/cloudbus/CloudBus.cpp:43:12: note: suggested alternative:
In file included from /home/rtadmin/RTafni/src/ACE_wrappers/ace/os_include/net/os_if.h:28:0,
                 from /home/rtadmin/RTafni/src/ACE_wrappers/ace/OS_NS_sys_socket.h:29,
                 from /home/rtadmin/RTafni/src/ACE_wrappers/ace/ACE.inl:8,
                 from /home/rtadmin/RTafni/src/ACE_wrappers/ace/ACE.h:886,
                 from /home/rtadmin/RTafni/src/ACE_wrappers/ace/String_Base.cpp:6,
                 from /home/rtadmin/RTafni/src/ACE_wrappers/ace/String_Base.h:882,
                 from /home/rtadmin/RTafni/src/ACE_wrappers/ace/SString.h:23,
                 from /home/rtadmin/RTafni/src/ACE_wrappers/ace/Service_Gestalt.h:26,
                 from /home/rtadmin/RTafni/src/ACE_wrappers/ace/Service_Object.h:27,
                 from /home/rtadmin/RTafni/src/ACE_wrappers/ace/Task.h:17,
                 from /home/rtadmin/RTafni/src/gadgetron/toolboxes/cloudbus/CloudBus.h:5,
                 from /home/rtadmin/RTafni/src/gadgetron/toolboxes/cloudbus/CloudBus.cpp:1:
/usr/include/net/if.h:197:29: note:   ‘if_nameindex’
 extern struct if_nameindex *if_nameindex (void) __THROW;
```
I can't find any reference of `if_nameindex` or `if_freenameindex` in the [ACE API documentation](http://www.dre.vanderbilt.edu/Doxygen/). Just removing the `ACE_OS::` namespace qualifier works, but I'm sure won't work on Windows. Any ideas?

**Comment by hansenms** [comment]
This code id never called on Windows anyway, it is in an #ifdef __linux__ anyway. I checked that it works on Ubuntu 14.04 without the ACE_OS and I have put up a PR (https://github.com/gadgetron/gadgetron/pull/117), feel free to merge it

**Comment by hansenms** [comment]
Solved in commit b0f008b21ee82a2667b0ec2cd17d3228c292e511
